File: Server/ElectricityBillPayment/ModelClass/DTO/Customer.cs
﻿using System;
using System.Collections.Generic;
using System.Text;

namespace ModelClass.DTO
{
    public class Customer
    {
        public Customer(){
            }
        public int CustomerId { get; set; }
        public int? UserId { get; set; }
        public int ZoneId { get; set; }
        public string CustomerName { get; set; }
        public string FatherName { get; set; }
        public string MotherName { get; set; }
        public string Email { get; set; }
        public string Password { get; set; }
        public string MobileNo { get; set; }
        public DateTime DateOfBirth { get; set; }
        public int Gender { get; set; }
        public int Religion { get; set; }
        public int Nationality { get; set; }
        public int BloodGroup { get; set; }
        public string Address { get; set; }
        public int CustomerType { get; set; }
        public double Latitude { get; set; }
        public double Longitude { get; set; }
        public string CreatedBy { get; set; }
        public DateTime CreatedDate { get; set; }
        public string UpdatedBy { get; set; }
        public DateTime? UpdatedDate { get; set; }
        public int? Status { get; set; }
        public string Image { get; set; }
        public  ICollection<MeterAssign> MeterAssign { get; set; }
        

    }

    public class CustomerSearch
    {
        public string Search { get; set; }
    }
}

File: Client/ElectricityBillClient/src/app/Model/Customer.ts
import { Zone } from './Zone';

export class Customer{
  CustomerId:number=0;
  UserId:number=0;
  ZoneId:number=0;
  CustomerName:string='';
  FatherName:string='';
  MotherName:string='';
  Email:string='';
  Password:string=''
  MobileNo:string=''
  DateOfBirth:Date|undefined;
  Gender:number=0;
  Religion:number=0;
  Nationality:number=0;
  BloodGroup:number=0;
  Address:string='';
  CustomerType:number=0;
  Latitude=0;
  Longitude=0;
  CreatedBy:string='';
  CreatedDate:Date|undefined;
  UpdatedBy:string='';
  UpdatedDate:Date|undefined;
  Status:number=0;
  Image:string='';

}

export class CustomerLocation{
  Latitude=0;
  Longitude=0;
}
